rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -48,38 +48,56 @@ get_node_names(
 
   DDS::DomainParticipant * participant = static_cast<ConnextNodeInfo *>(node->data)->participant;
   DDS::InstanceHandleSeq handles;
+
   if (participant->get_discovered_participants(handles) != DDS::RETCODE_OK) {
     RMW_SET_ERROR_MSG("unable to fetch discovered participants.");
     return RMW_RET_ERROR;
   }
+
+  DDS::DomainParticipantQos participant_qos;
+  DDS::ReturnCode_t status = participant->get_qos(participant_qos);
+
+  if (status != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("failed to get default participant qos");
+    return RMW_RET_ERROR;
+  }
+
+  rmw_ret_t final_ret = RMW_RET_OK;
+
   auto length = handles.length() + 1;  // add yourself
+
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
-  rcutils_ret_t rcutils_ret = rcutils_string_array_init(node_names, length, &allocator);
+
+  // Pre-allocate temporary buffer for all node names & namespaces
+  // Nodes that are not created by ROS 2 could provide empty names
+  // Such names should not be returned
+  rcutils_string_array_t tmp_names_list = rcutils_get_zero_initialized_string_array();
+  rcutils_string_array_t tmp_namespaces_list = rcutils_get_zero_initialized_string_array();
+
+  int named_nodes_num = 1;
+
+  rcutils_ret_t rcutils_ret = rcutils_string_array_init(&tmp_names_list, length, &allocator);
+
   if (rcutils_ret != RCUTILS_RET_OK) {
     RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
-    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+    final_ret = rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+    goto cleanup;
   }
-  rcutils_ret = rcutils_string_array_init(node_namespaces, length, &allocator);
+
+  rcutils_ret = rcutils_string_array_init(&tmp_namespaces_list, length, &allocator);
+
   if (rcutils_ret != RCUTILS_RET_OK) {
     RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
-    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+    final_ret = rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+    goto cleanup;
   }
 
-  DDS::DomainParticipantQos participant_qos;
-  DDS::ReturnCode_t status = participant->get_qos(participant_qos);
-  if (status != DDS::RETCODE_OK) {
-    RMW_SET_ERROR_MSG("failed to get default participant qos");
-    return RMW_RET_ERROR;
-  }
-  node_names->data[0] = rcutils_strdup(participant_qos.participant_name.name, allocator);
-  if (!node_names->data[0]) {
-    RMW_SET_ERROR_MSG("could not allocate memory for node name");
-    return RMW_RET_BAD_ALLOC;
-  }
-  node_namespaces->data[0] = rcutils_strdup(node->namespace_, allocator);
-  if (!node_names->data[0]) {
-    RMW_SET_ERROR_MSG("could not allocate memory for node namespace");
-    return RMW_RET_BAD_ALLOC;
+  // add yourself
+  tmp_names_list.data[0] = rcutils_strdup(participant_qos.participant_name.name, allocator);
```

Comment:
This only populates the `name` bub not the `namespace` of the current node. As a result a test in `rcl` is failing since this PR was merged.

The CI builds ran for this change also didn't cover `rcl` as far as I can see. Please address this regression as soon as possible.

Reply:
My mistake; I only ran test for this package and test_communication. I'll fix this.

Reply:
See https://github.com/ros2/rmw_connext/pull/384 for a fix.